# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/txmgr/TransactionItem.cpp
# Commit: d3816f82c0b9
# Full Hash: d3816f82c0b9d4ceca0bb2ad177694164ed0921f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-03 21:40:10
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 2: Mark transaction class methods and their caller methods as `MOZ_CAN_RUN_SCRIPT r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69303
# ==============================================================================

diff -r 876b1031a6ae -r d3816f82c0b9 editor/txmgr/TransactionItem.cpp
--- a/editor/txmgr/TransactionItem.cpp	Fri Apr 03 01:38:22 2020 +0000
+++ b/editor/txmgr/TransactionItem.cpp	Fri Apr 03 01:39:36 2020 +0000
@@ -6,6 +6,7 @@
 #include "TransactionItem.h"
 
 #include "mozilla/mozalloc.h"
+#include "mozilla/OwningNonNull.h"
 #include "mozilla/TransactionManager.h"
 #include "mozilla/TransactionStack.h"
 #include "nsCOMPtr.h"
@@ -76,7 +77,8 @@
   if (!mTransaction) {
     return NS_OK;
   }
-  nsresult rv = mTransaction->DoTransaction();
+  OwningNonNull<nsITransaction> transaction = *mTransaction;
+  nsresult rv = transaction->DoTransaction();
   NS_WARNING_ASSERTION(NS_SUCCEEDED(rv),
                        "nsITransaction::DoTransaction() failed");
   return rv;
@@ -97,7 +99,8 @@
     return NS_OK;
   }
 
-  rv = mTransaction->UndoTransaction();
+  OwningNonNull<nsITransaction> transaction = *mTransaction;
+  rv = transaction->UndoTransaction();
   if (NS_SUCCEEDED(rv)) {
     return NS_OK;
   }
@@ -257,6 +260,8 @@
 
 nsresult TransactionItem::RecoverFromRedoError(
     TransactionManager* aTransactionManager) {
+  OwningNonNull<nsITransaction> transaction = *mTransaction;
+
   // If this method gets called, we already successfully called
   // RedoTransaction() for the transaction item itself. Undo all
   // the children that successfully called RedoTransaction(),
@@ -271,7 +276,7 @@
     return NS_OK;
   }
 
-  rv = mTransaction->UndoTransaction();
+  rv = transaction->UndoTransaction();
   NS_WARNING_ASSERTION(NS_SUCCEEDED(rv),
                        "nsITransaction::UndoTransaction() failed");
   return rv;